cupy/core/core.pyx

```diff
@@ -2257,7 +2266,7 @@ cdef class broadcast:
                 a_sh = a._shape[a_ndim - i - 1]
                 if a_sh == r_shape[i]:
                     r_strides[i] = a._strides[a_ndim - i - 1]
-                elif a_sh != 1:
+                elif a_sh > 1:
```

Comment:
Why is the change needed?

Reply:
Because `a_sh` will be `0` if all the corresponding dimensions of the arrays are either 0 or 1.

Reply:
I got your intension.

I found `python -m unittest tests/cupy_tests/manipulation_tests/test_dims.py` passes with `elif a_sh != 0:`, too.  If `a_sh == 0` then it seems `a_sh == r_shape[i]` holds.

Reply:
It seems you're right. Thank you for pointing out.